src/unittest/test_utilities.cpp

```diff
@@ -636,3 +638,69 @@ void TestUtilities::testSanitizeDirName()
 	UASSERT(sanitizeDirName("cOnIn$", "~") == "~cOnIn$");
 	UASSERT(sanitizeDirName(" cOnIn$ ", "~") == "_cOnIn$_");
 }
+
+template <typename F, typename C>
```

Comment:
Something like this might be more readable
```suggestion
template <typename FUN_T, typename VALUE_T>
```